controlplane/pkg/nsmd/network_service_server.go

```diff
@@ -158,19 +158,20 @@ func (srv *networkServiceServer) createLocalNSERequest(endpoint *registry.NSEReg
 	return message
 }
 
-func getEndpoint(srv *networkServiceServer, request *networkservice.NetworkServiceRequest) (*registry.NSERegistration, error) {
-	endpoints := srv.model.GetNetworkServiceEndpoints(request.GetConnection().GetNetworkService())
+func (srv *networkServiceServer) getEndpoint(request *networkservice.NetworkServiceRequest) (*registry.NSERegistration, error) {
+	endpoint, err := srv.model.SelectEndpoint(request.GetConnection().GetNetworkService())
```

Comment:
Could we incorporate remote endpoints into SelectEndpoints... we really do want to do *one* selection, whether remote or local.  For right now, that's going to be round robin, but @nickolaev is working on more sophisticated selection.  So in that case, SelectEndpoint would need to make the call the the registry to get its list of candidates, and round robin among them.

Reply:
Sure, this make seanse

Reply:
On the other side, Model should not do any remote calls.

Reply:
I agree with you, but it might be even better to say that model doesn't make any calls that leave the process (whether to another nsmd or to an NSE).